src/metabase/driver.clj

```diff
@@ -86,6 +86,7 @@
 (defn do-with-driver
   "Impl for `with-driver`."
   [driver f]
+  {:pre [(keyword? driver)]}
```

Comment:
we're doing this here instead of at `with-driver` is because `driver` for `with-driver` could be a symbol.

For example `with-drivers` is expanded into `(doseq [driver drivers] (with-driver driver))` where driver is a symbol.

That's said I think putting here is fine, we still get an error when someone trying to use it with a set/seq, it's just not easy to figure out what's the error is about if it's put in `with-driver`.